### PR TITLE
Use static methods inside data providers

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | n/a
| License       | MIT
| Doc PR        | n/a

refs https://github.com/symfony/symfony/pull/48668

follows
* https://github.com/symfony/symfony/pull/48625

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Tests/Config/ContainerParametersResourceCheckerTest.php [hunk 1]
```diff
@@ -62,11 +62,11 @@ public function isFreshProvider()
         }, false];
 
         yield 'fresh on every identical parameters' => [function (MockObject $container) {
-            $container->expects($this->exactly(2))->method('hasParameter')->willReturn(true);
-            $container->expects($this->exactly(2))->method('getParameter')
+            $container->expects(self::exactly(2))->method('hasParameter')->willReturn(true);
+            $container->expects(self::exactly(2))->method('getParameter')
```
Comment (OskarStark) [inline src/Symfony/Component/DependencyInjection/Tests/Config/ContainerParametersResourceCheckerTest.php:66]: All these methods are defined as static anyway, shall we fix them through the whole codebase starting from 5.4 @nicolas-grekas ? Automated via rector ofc

File: src/Symfony/Component/Finder/Tests/Iterator/SizeRangeFilterIteratorTest.php [hunk 2]
```diff
@@ -41,7 +41,7 @@ public function getAcceptData()
         ];
 
         return [
-            [[new NumberComparator('< 1K'), new NumberComparator('> 0.5K')], $this->toAbsolute($lessThan1KGreaterThan05K)],
+            [[new NumberComparator('< 1K'), new NumberComparator('> 0.5K')], self::toAbsolute($lessThan1KGreaterThan05K)],
```
Comment (OskarStark) [inline src/Symfony/Component/Finder/Tests/Iterator/SizeRangeFilterIteratorTest.php:44]: Method was already declared `static`

File: src/Symfony/Component/DependencyInjection/Tests/Config/ContainerParametersResourceCheckerTest.php [hunk 3]
```diff
@@ -62,11 +62,11 @@ public function isFreshProvider()
         }, false];
 
         yield 'fresh on every identical parameters' => [function (MockObject $container) {
-            $container->expects($this->exactly(2))->method('hasParameter')->willReturn(true);
-            $container->expects($this->exactly(2))->method('getParameter')
+            $container->expects(self::exactly(2))->method('hasParameter')->willReturn(true);
```
Comment (derrabus) [inline src/Symfony/Component/DependencyInjection/Tests/Config/ContainerParametersResourceCheckerTest.php:65]: You've changed the calls inside this method to static, but don't we need to change the declaration of `isFreshProvider()` to `static` as well?

File: src/Symfony/Component/Finder/Tests/GitignoreTest.php [hunk 4]
```diff
@@ -394,7 +394,7 @@ public function provider(): array
 
     public function providerExtended(): array
```
Comment (derrabus) [inline src/Symfony/Component/Finder/Tests/GitignoreTest.php:395]:
```suggestion
    public static function providerExtended(): array
```

File: src/Symfony/Component/VarDumper/Tests/Caster/CasterTest.php [hunk 5]
```diff
@@ -38,12 +38,12 @@ class CasterTest extends TestCase
...
         } else {
-            $filteredArray = Caster::filter($this->referenceArray, $filter, $listedProperties);
+            $filteredArray = Caster::filter(self::$referenceArray, $filter, $listedProperties);
         }
 
-        $this->assertSame($expectedDiff, array_diff_assoc($this->referenceArray, $filteredArray));
+        $this->assertSame($expectedDiff, array_diff_assoc(self::$referenceArray, $filteredArray));
     }
 
     public function provideFilter()
```
Comment (derrabus) [inline src/Symfony/Component/VarDumper/Tests/Caster/CasterTest.php:49]:
```suggestion
    public static function provideFilter()
```

File: src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToRfc3339TransformerTest.php [hunk 6]
```diff
@@ -51,12 +51,12 @@ public function allProvider()
 
     public function transformProvider()
```
Comment (derrabus) [inline src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToRfc3339TransformerTest.php:52]:
```suggestion
    public static function transformProvider()
```

File: src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToRfc3339TransformerTest.php [hunk 7]
```diff
@@ -51,12 +51,12 @@ public function allProvider()
...
     {
-        return $this->allProvider();
+        return self::allProvider();
     }
 
     public function reverseTransformProvider()
```
Comment (derrabus) [inline src/Symfony/Component/Form/Tests/Extension/Core/DataTransformer/DateTimeToRfc3339TransformerTest.php:57]:
```suggestion
    public static function reverseTransformProvider()
```

File: src/Symfony/Component/Form/Tests/DependencyInjection/FormPassTest.php [hunk 8]
```diff
@@ -285,31 +285,31 @@ public function privateTaggedServicesProvider()
                 function (ContainerBuilder $container) {
                     $formTypes = $container->getDefinition('form.extension')->getArgument(0);
 
-                    $this->assertInstanceOf(Reference::class, $formTypes);
+                    self::assertInstanceOf(Reference::class, $formTypes);
```
Comment (derrabus) [inline src/Symfony/Component/Form/Tests/DependencyInjection/FormPassTest.php:288]: `privateTaggedServicesProvider()` is not `static`

File: src/Symfony/Component/DependencyInjection/Tests/Config/ContainerParametersResourceCheckerTest.php (hunk 3 above)
Comment (OskarStark) [reply src/Symfony/Component/DependencyInjection/Tests/Config/ContainerParametersResourceCheckerTest.php:65]: Forgot to push 😄

File: src/Symfony/Component/Form/Tests/Command/DebugCommandTest.php [hunk 9]
```diff
@@ -215,7 +215,7 @@ public function provideCompletionSuggestions(): iterable
 
         yield 'form_type' => [
             [''],
-            $this->getCoreTypes(),
+            self::getCoreTypes(),
```
Comment (derrabus) [inline src/Symfony/Component/Form/Tests/Command/DebugCommandTest.php:218]: `provideCompletionSuggestions()` is not `static`.

File: src/Symfony/Component/Finder/Tests/Iterator/SizeRangeFilterIteratorTest.php (hunk 2 above)
Comment (derrabus) [reply src/Symfony/Component/Finder/Tests/Iterator/SizeRangeFilterIteratorTest.php:44]: `getAcceptData()` is not `static`.

**Review by derrabus — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Finder/Tests/Iterator/SizeRangeFilterIteratorTest.php (hunk 2 above)
Comment (OskarStark) [reply src/Symfony/Component/Finder/Tests/Iterator/SizeRangeFilterIteratorTest.php:44]: Please check the PR header, those gets all fixed automatically via rector, these are just things were we have to add manuall work

Comment (derrabus) [reply src/Symfony/Component/Finder/Tests/Iterator/SizeRangeFilterIteratorTest.php:44]: Oh, I see. Would it make sense to push these manual changes as a separate commit to #48668? Other than that, this PR is fine as far as I can tell.

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Finder/Tests/Iterator/SizeRangeFilterIteratorTest.php (hunk 2 above)
Comment (OskarStark) [reply src/Symfony/Component/Finder/Tests/Iterator/SizeRangeFilterIteratorTest.php:44]: I want to bring the manual stuff beside this PR into the codebase, unmerge everything and then apply the automatic changes branch by branch.

**Comment by OskarStark** [comment]
RTM from my side as another partial one

File: src/Symfony/Component/DependencyInjection/Tests/Config/ContainerParametersResourceCheckerTest.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Tests/Config/ContainerParametersResourceCheckerTest.php:66]: No need to. That wouldn't provide any real added value but would consume a lot of time for sure.

**Comment by nicolas-grekas** [comment]
Thank you @OskarStark.

**Pull request merged**